src/v/cluster/controller_backend.cc

```diff
@@ -863,6 +885,24 @@ ss::future<> controller_backend::remove_from_shard_table(
       });
 }
 
+ss::future<std::error_code> controller_backend::create_materialized(
+  model::ntp ntp, model::revision_id rev) {
+    auto cfg = _topics.local().get_topic_cfg(model::topic_namespace_view(ntp));
+    if (!cfg) {
+        // partition was already removed, do nothing
+        co_return errc::success;
+    }
+    if (!cfg->properties.source_topic) {
+        // State has been confused, this method should only be called on a topic
+        // that has been marked as materialized
+        co_return errc::topic_invalid_config;
```

Comment:
This will prevent us from making any further progress. I think it should be `vassert` really.

Reply:
Good point!